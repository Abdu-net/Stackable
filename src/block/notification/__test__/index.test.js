import { edit, save } from '../index'

describe( 'Notification', () => {
	test( 'block edit matches snapshots', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				text: 'This is an informational alert, usually used for successful subscriptions, promo announcements, and the like.',
				color: '#000000',
				textColor: '#ffffff',
				notifType: 'success',
				dismissible: false,
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( 'block save matches snapshots', () => {
		const wrapper = save( {
			isSelected: false,
			attributes: {
				text: 'This is an informational alert, usually used for successful subscriptions, promo announcements, and the like.',
				color: '#000000',
				textColor: '#ffffff',
				notifType: 'success',
				dismissible: false,
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )
} )
