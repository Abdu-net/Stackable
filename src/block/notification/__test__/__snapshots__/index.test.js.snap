// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification block edit matches snapshots 1`] = `
<Fragment>
  <InspectorControls>
    <PanelBody>
      <ToggleControl
        checked={false}
        label="Dismissible"
        onChange={[Function]}
      />
      <SelectControl
        label="Notification Type"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Success",
              "value": "success",
            },
            Object {
              "label": "Error",
              "value": "error",
            },
            Object {
              "label": "Warning",
              "value": "warning",
            },
            Object {
              "label": "Information",
              "value": "info",
            },
          ]
        }
        value="success"
      />
    </PanelBody>
    <PanelColorSettings
      colorSettings={
        Array [
          Object {
            "label": "Background Color",
            "onChange": [Function],
            "value": "#000000",
          },
          Object {
            "label": "Text Color",
            "onChange": [Function],
            "value": "#ffffff",
          },
        ]
      }
      title="Color Settings"
    />
  </InspectorControls>
  <div
    className="ugb-notification type-success dismissible-false"
  >
    <RichText
      className="wp-ugb-notif notif-success"
      onChange={[Function]}
      placeholder={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "color": "#ffffff",
        }
      }
      tagName="p"
      value="This is an informational alert, usually used for successful subscriptions, promo announcements, and the like."
    />
  </div>
</Fragment>
`;

exports[`Notification block save matches snapshots 1`] = `
<div
  className="ugb-notification type-success dismissible-false"
  data-uid="14f948"
>
  <Unknown
    className="wp-ugb-notif notif-success"
    style={
      Object {
        "backgroundColor": "#000000",
        "color": "#ffffff",
      }
    }
    tagName="p"
    value="This is an informational alert, usually used for successful subscriptions, promo announcements, and the like."
  />
</div>
`;
